test_data/rules/no_adjacent_strings_in_list.dart

```diff
@@ -5,23 +5,45 @@
 // test w/ `dart test -N no_adjacent_strings_in_list`
 
 void bad() {
-  List<String> list = <String>[
+  var list = [
     'a' // LINT
     'b',
     'c',
   ];
 
-  List<String> list2 = <String>[
+  var list2 = [
     'a' // LINT
     'b'
     'c'
   ];
+
+  var list3 = [
```

Comment:
Consider using reflective tests for the new tests.

Reply:
I was thinking it would be more confusing (like, really confusing) to have the test cases split between integration tests and reflective tests.

But I am really on board with moving to reflective tests ASAP.

Reply:
I'm fine with you not changing it, but the reason I said something is that we've been writing new tests as reflective tests so that we don't increase the technical debt in the process of writing new tests. I understand that having both styles is confusing, but I think that not increasing the technical debt is more important.